# Hypothesis testing problems

## Exercise 1

You are a nutritionist researching two different types of diets to see if there is a significant difference in weight loss after one month. You choose two random groups of people, one group follows the first diet and the other follows the second. At the end of the month, you record the weight loss (in kg) for each person:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

With these data, it seeks to answer the following question: is there a significant difference in average weight loss between people who followed the first diet and those who followed the second diet?

To draw conclusions, follow the points below:

- State the hypothesis: null and alternative hypothesis.
- Perform the test to test the hypothesis. You can use a Student's t-test.
- Analyze the conclusions.

## Through hypothesis testing for the mean
#### Diet 1 are the population
#### Diet 2 is the sample

In [11]:
import numpy as np

d1 = [2.0,2.5,3.0,2.8,2.9,2.7,2.5]
d2 = [3.0,3.2,3.1,2.9,2.8,3.0,3.2]
a = 0.05

mu = np.mean(d1)
print(mu)

2.628571428571429


##### H_0= The mean of weight loss is 2.62
##### H_1= The mean of weight loss is not 2.62
##### accuracy a=0.05

In [12]:
# TODO
from scipy.stats import ttest_1samp
stat, p_value = ttest_1samp(d2, mu)
print(p_value)

if p_value < a:
    print('The mean os weight loss is not 2.62')
else:
    print('The mean os weight loss is 2.62')

0.00039995168632541475
The mean os weight loss is not 2.62


### Through comparison of two samples

##### H_0=The weight loss are the same with both diets
##### H_1=The weight loss are not the same with both diets

In [13]:
from scipy.stats import ttest_ind
stat, p_value = ttest_ind(d2, d1, equal_var=False)# not assume that the variance of both samples are the same
print(p_value)

if p_value < a:
    print('The mean os weight loss is not 2.62')
else:
    print('The mean os weight loss is 2.62')

0.01965630735431519
The mean os weight loss is not 2.62


## ANOVA

**ANOVA** (*Analysis of Variance*) is a statistical technique used to compare the measures of two or more groups. The idea behind ANOVA is to decompose the total variability in the data into two components: between-group variability and within-group variability:

- **Between-group variability**: This variability refers to the differences between the group means. If this variability is considerably larger than the within-group variability, it could be an indication that at least one of the group means is different.
- **Within-group variability**: This variability refers to the dispersion of the data within each group. If all groups have similar variability, then any noticeable difference in group means could be considered significant.

Hypotheses in ANOVA typically are:

- **Null hypothesis** ($H₀$): The means of all groups are equal.
- **Alternative hypothesis** ($H₁$): At least one of the group means is different.

If the ANOVA test result is significant (e.g., a p-value less than a threshold such as 0.05), this suggests that at least one group mean is different.

### Exercise 2

A farmer decides to test three different types of fertilizers to determine if one is superior in terms of corn production. The farmer plants corn on 15 identical plots and uses all three fertilizers (5 plots for each type). At the end of the season, he measures the corn yield (in kg) of each plot, with the following result:

| Fertilizer 1 | Fertilizer 2 | Fertilizer 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

With this data, he seeks to answer the following question: is there a significant difference in average corn yield between the three types of fertilizers?

To help you, follow the points below:

- State the hypothesis: null and alternative hypothesis.
- Perform the ANOVA test.
- Analyze the conclusions.
- If one fertilizer is better than another, how can we know it?

In [14]:
# TODO
from scipy.stats import f_oneway

D1 = [20,21,20,19,20]
D2 = [22,21,23,22,21]
D3 = [24,23,22,23,24]

stat, p_value = f_oneway(D1, D2, D3)
if p_value < a:
    print('There are significant differences between at least two of the samples')
else:
    print('No hay diferencias significativas entre las muestras')

There are significant differences between at least two of the samples


##### Like the are significant differences between the samples, there must be one fertilizer better than another
To know which one is the best fertilizer, we can compare using T-Test two by two, in all possible combination.
Then, the one with the smaller p_value is the one with larger differences in means, so it will be the one with higher corn production

One method to make this comparison is by Tukey's test

In [16]:
from statsmodels.stats import multicomp 

D = D1 + D2 + D3

labels = ['Fertilizer 1'] * len(D1) + ['Fertilizer 2'] * len(D2) + ['Fertilizer 3'] * len(D3)
tukey = multicomp.pairwise_tukeyhsd(D, labels)

print(tukey)

     Multiple Comparison of Means - Tukey HSD, FWER=0.05      
   group1       group2    meandiff p-adj  lower  upper  reject
--------------------------------------------------------------
Fertilizer 1 Fertilizer 2      1.8 0.0099 0.4572 3.1428   True
Fertilizer 1 Fertilizer 3      3.2 0.0001 1.8572 4.5428   True
Fertilizer 2 Fertilizer 3      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------------------


The fertilizer 2 is producing on average 1.8 units more of corn. 

The fertilizer 3 is producing on average 3.2 units more of corn.

The fertilizer 3 is producing on average 1.4 units more of corn.

#### The fertilizer 3 is the best because it is producing on average more corn than fertilizer 1 and fertilizer 2. 